File: ALDS/2_A.js
'use strict';

function Main(input) {
	input = input.split('\n');
	const N = Number(input[0]);
	const A = input[1].split(' ').map(Number);
	let c = 0;
	function bubbleSort(A, N) {
		let f = true;
		while (f) {
			f = false;
			for (let i = N - 1; i > 0; i--) {
				if (A[i] < A[i - 1]) {
					[A[i], A[i - 1]] = [A[i - 1], A[i]];
					c++;
					f = true;
				}
			}
		}
		return A;
	}
	console.log(bubbleSort(A, N).join(' '));
	console.log(c);
}

Main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: ALDS1/2_A.py
N = int(input())
A = list(map(int, input().split()))

flag = 1
c = 0
while flag:
    flag = 0
    for j in range(N - 1, 0, -1):
        if A[j] < A[j - 1]:
            c += 1
            A[j], A[j - 1] = A[j - 1], A[j]
            flag = 1

print(" ".join([str(a) for a in A]))
print(c)
